coffeecard/CoffeeCard.WebApi/Logging/Enrichers/UserIdEnricher.cs

```diff
@@ -0,0 +1,40 @@
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace CoffeeCard.WebApi.Logging.Enrichers
+{
+    public class UserIdEnricher : ILogEventEnricher
+    {
+        private const string UserIdPropertyName = "UserId";
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        public UserIdEnricher() : this(new HttpContextAccessor())
+        {
+        }
+
+        private UserIdEnricher(IHttpContextAccessor contextAccessor)
+        {
+            _contextAccessor = contextAccessor;
+        }
+
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            if (_contextAccessor.HttpContext == null)
+                return;
+
+            var userIdProperty = new LogEventProperty(UserIdPropertyName, new ScalarValue(GetUserId()));
+
+            logEvent.AddOrUpdateProperty(userIdProperty);
+        }
+
+        private string GetUserId()
+        {
+            var claims = _contextAccessor.HttpContext.User.Claims.ToList();
+            var id = claims.Count > 2 ? claims[2].Value : "NONE";
+            
+            return $"userid:{id}";
```

Comment:
It would be better to not add the enricher in the log statement if no user is logged in, rather than adding `NONE`. Is that possible?